homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1037,196 +952,6 @@ async def async_set_led_brightness(self, brightness: int = 2):
-            self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER
-            self._speed_list = [
-                mode.name
-                for mode in AirhumidifierOperationMode
-                if mode is not AirhumidifierOperationMode.Auto
-            ]
-
-        self._state_attrs.update(
-            {attribute: None for attribute in self._available_attributes}
-        )
-
-    async def async_update(self):
-        """Fetch state from the device."""
-        # On state change the device doesn't provide the new state immediately.
-        if self._skip_update:
-            self._skip_update = False
-            return
-
-        try:
-            state = await self.hass.async_add_executor_job(self._device.status)
-            _LOGGER.debug("Got new state: %s", state)
-
-            self._available = True
-            self._state = state.is_on
-            self._state_attrs.update(
-                {
-                    key: self._extract_value_from_attribute(state, value)
-                    for key, value in self._available_attributes.items()
-                }
-            )
-
-        except DeviceException as ex:
-            if self._available:
-                self._available = False
-                _LOGGER.error("Got exception while fetching the state: %s", ex)
-
-    @property
-    def speed_list(self) -> list:
-        """Get the list of available speeds."""
-        return self._speed_list
-
-    @property
-    def speed(self):
-        """Return the current speed."""
-        if self._state:
-            return AirhumidifierOperationMode(self._state_attrs[ATTR_MODE]).name
-
-        return None
-
-    async def async_set_speed(self, speed: str) -> None:
-        """Set the speed of the fan."""
-        if self.supported_features & SUPPORT_SET_SPEED == 0:
-            return
-
-        _LOGGER.debug("Setting the operation mode to: %s", speed)
-
-        await self._try_command(
-            "Setting operation mode of the miio device failed.",
-            self._device.set_mode,
-            AirhumidifierOperationMode[speed.title()],
-        )
-
-    async def async_set_led_brightness(self, brightness: int = 2):
-        """Set the led brightness."""
-        if self._device_features & FEATURE_SET_LED_BRIGHTNESS == 0:
-            return
-
-        await self._try_command(
-            "Setting the led brightness of the miio device failed.",
-            self._device.set_led_brightness,
-            AirhumidifierLedBrightness(brightness),
-        )
-
-    async def async_set_target_humidity(self, humidity: int = 40):
-        """Set the target humidity."""
-        if self._device_features & FEATURE_SET_TARGET_HUMIDITY == 0:
-            return
-
-        await self._try_command(
-            "Setting the target humidity of the miio device failed.",
-            self._device.set_target_humidity,
-            humidity,
-        )
-
-    async def async_set_dry_on(self):
```

Comment:
How do we support dry mode, led brightness and motor speed services now?

Reply:
The question is whether we should use services or maybe entities (switch for dry mode, select for led brightness, etc.). Motor speed needs a service for sure.

Reply:
I think we using entities to switch or control will be better, we could add those later. _Let try to make this PR to big._ For now I have only removed the humidifier code from fan.py to make merging of the fan related PR easier.
We could add the additional entities for motor_speed, led_brightness, dry_mode, buzzer and child_lock later. Finally we should remove _Humidifier only_ services since they cannot be used any more. I will remove the AIRHUMIDIFIER related feature flags and FEATURES for now. What services need to be removed that are AIRHUMIDIFIER  specific?

Reply:
Yes, I totally agree that adding another platforms should be in the next PRs.
I think those services are for humidifiers only and should be removed from `fan.py`: `SERVICE_SET_TARGET_HUMIDITY`, `SERVICE_SET_DRY_ON`, `SERVICE_SET_DRY_OFF`, `SERVICE_SET_MOTOR_SPEED`

Reply:
`dry_mode`, `child_lock` and `buzzer` could later be implemented as `switch` https://www.home-assistant.io/integrations/switch/. Implementing `switch.turn_on` and `switch.turn_off` will remove the need for specific services.
Motor control could be implemented using `input_number` https://www.home-assistant.io/integrations/input_number/
This will also allow using a slider.
The led brightness could be implemented using an input select: https://www.home-assistant.io/integrations/input_select/.

Reply:
I think `input_select` and `input_number` can't be used by integrations, but maybe I'm wrong.
We have now `select` platform https://github.com/home-assistant/core/pull/51849

Reply:
Removed not supported services.